graph/pkg/identity/ldap.go

```diff
@@ -84,12 +84,165 @@ func NewLDAPBackend(lc ldap.Client, config config.LDAP, logger *log.Logger) (*LD
 	}, nil
 }
 
-func (i *LDAP) GetUser(ctx context.Context, userID string) (*libregraph.User, error) {
-	i.logger.Debug().Str("backend", "ldap").Msg("GetUser")
-	userID = ldap.EscapeFilter(userID)
+// CreateUser implements the Backend Interface. It converts the libregraph.User into an
+// LDAP User Entry (using the inetOrgPerson LDAP Objectclass) add adds that to the
+// configured LDAP server
+func (i *LDAP) CreateUser(ctx context.Context, user libregraph.User) (*libregraph.User, error) {
+
+	ar := ldap.AddRequest{
+		DN: fmt.Sprintf("uid=%s,%s", *user.OnPremisesSamAccountName, i.userBaseDN),
+		Attributes: []ldap.Attribute{
+			{
+				Type: "objectClass",
+				Vals: []string{"inetOrgPerson", "organizationalPerson", "person", "top"},
+			},
+			// inetOrgPerson requires "cn"
+			{
+				Type: "cn",
```

Comment:
yeah ... so inetOrgPerson or ... whatever schema the ldap server uses. in reva defined a struct that is uset for all mapping. with a set of defaults: https://github.com/cs3org/reva/blob/master/pkg/user/manager/ldap/ldap.go#L63-L89

I guess for now we can hardcode this and extract a common ldap package after GA? Not 'basic go ldap functionality' but a 'common ldap configuration functionality' package?

Reply:
See: https://github.com/owncloud/ocis/blob/master/graph/pkg/identity/ldap.go#L31 

And again my point about we strict wrt to supported schema of the write part of the API